### PR TITLE
[#897] Feature/contacts approval flow

### PR DESCRIPTION
**Pull request by vaszig**

### CODE REVIEW

File: src/open_inwoner/accounts/views/contacts.py
Comment (Bartvaderkin) [inline src/open_inwoner/accounts/views/contacts.py:161]: I think it is convention to not use raw POST but always try to go for a Form, and use Choices instead of magic strings so validation is a bit harder and more regular. 

The two buttons can have the same name in the HTML but different values and then the Form is just a CharField with the choices for the incoming data.

Also, looking at the post method: what happens if there is no approved or rejected? It could `raise BadRequest` since there is no HTML for this view.

**Review by Bartvaderkin — CHANGES_REQUESTED** [review]
I put my notes in the code. 

I also noticed this that might existed before: maybe these buttons could be closer? The Accept button is really far from the Reject, and the Reject button label says Terug (Back)

Screenshot from 2022-12-07 14-15-50

File: src/open_inwoner/templates/pages/profile/contacts/list.html
Comment (Bartvaderkin) [inline src/open_inwoner/templates/pages/profile/contacts/list.html:64]: (this is a lot.. but then I don't know how to do it differently without making it worse so meh :wink: )

**Comment by Bartvaderkin** [comment]
I forgot to mention this: there is a small CSS bug in the first header of the contacts table (probably from earlier):

image

**Comment by vaszig** [comment]
> I put my notes in the code.
> 
> I also noticed this that might existed before: maybe these buttons could be closer? The Accept button is really far from the Reject, and the Reject button label says Terug (Back)
> 
> Screenshot from 2022-12-07 14-15-50

Yes, this was there before I touch the frontend!It seems weird but if @alextreme wants I can create a task for it.

**Comment by vaszig** [comment]
> I forgot to mention this: there is a small CSS bug in the first header of the contacts table (probably from earlier):
> 
> image

Yes, there is a task in taiga about this. A change in the cases tables has broken all tables.

File: src/open_inwoner/accounts/views/contacts.py
Comment (vaszig) [reply src/open_inwoner/accounts/views/contacts.py:161]: Discussed this with Anna as well and I decided to go for the same approach as the one in deleting a contact. That's why you saw the ids in the for loop.

File: src/open_inwoner/accounts/models.py
Comment (vaszig) [inline src/open_inwoner/accounts/models.py:238]: I removed this method as it complicated things when we want to refer to different kind of contacts.

File: src/open_inwoner/accounts/views/contacts.py
Comment (Bartvaderkin) [reply src/open_inwoner/accounts/views/contacts.py]: Ok, if it's same pattern as the other and Anna is ok with it then it's fine.

Comment (vaszig) [reply src/open_inwoner/accounts/views/contacts.py]: Let's wait for @annashamray 's approval as well because she hasn't seen this implemented.

**Comment by alextreme** [comment]
> > I put my notes in the code.
> > I also noticed this that might existed before: maybe these buttons could be closer? The Accept button is really far from the Reject, and the Reject button label says Terug (Back)
> > Screenshot from 2022-12-07 14-15-50
> 
> Yes, this was there before I touch the frontend!It seems weird but if @alextreme wants I can create a task for it.

Task #960 created, noticed it before so lets make this prettier.

File: src/open_inwoner/scss/components/Contacts/Contacts.scss
Comment (vaszig) [reply src/open_inwoner/scss/components/Contacts/Contacts.scss:42]: Since it's a button class, I followed the existing pattern in the project (button--danger, button--disabled etc.) and removed this rule from contacts scss file. I added it to the button.scss.

File: src/open_inwoner/accounts/views/contacts.py
Comment (annashamray) [inline src/open_inwoner/accounts/views/contacts.py:48]: it looks excessive. Why not to have just `context["pending_approval_list"] = pending_approvals` and check the count in the template?

File: src/open_inwoner/templates/pages/profile/contacts/list.html
Comment (annashamray) [inline src/open_inwoner/templates/pages/profile/contacts/list.html:16]: I don't see this sentence it in the design (https://projects.invisionapp.com/share/Y211V6JRHPC4#/screens/470424529). It the design updated somewhere else?

File: src/open_inwoner/templates/pages/profile/contacts/list.html
Comment (annashamray) [inline src/open_inwoner/templates/pages/profile/contacts/list.html:27]: I don't think the value is needed here. I think the button name is enough (and in the view you can just check something like `"contact_reject" in request.POST`

File: src/open_inwoner/templates/pages/profile/contacts/list.html
Comment (annashamray) [inline src/open_inwoner/templates/pages/profile/contacts/list.html:24]: I could be dryer (with moving this form into a separate template tag and reusing it in the loop below), but not necessary though

File: src/open_inwoner/scss/components/Contacts/Contacts.scss
Comment (annashamray) [inline src/open_inwoner/scss/components/Contacts/Contacts.scss:2]: Hmm I don't see why it should be flex

File: src/open_inwoner/scss/components/Contacts/Contacts.scss
Comment (annashamray) [inline src/open_inwoner/scss/components/Contacts/Contacts.scss:9]: You include this padding into all child tags of `.approval`. Whi not to change the padding of the parent class itself?

File: src/open_inwoner/accounts/views/contacts.py
Comment (annashamray) [reply src/open_inwoner/accounts/views/contacts.py]: I think there is a little misunderstanding. 
I said that there is no need in adding additional forms into `ContactListView` context - we can just generate forms in the template since it's only csrf token, two buttons and 1 action. It's how the delete buttons for contacts are done. 
Security-wise it should be ok, since we send csrf token. 

But it's a different story with the view which process POST request (`ContactApprovalView`). 
Here we can use forms to escape and validate user input, especially if we use values in buttons. We can use only button names like django admin does, and they do it without extra validation. I don't know is it a best practice or not.

**Review by annashamray — COMMENTED** [review]
Nice! I'll approve after CI is happy

File: src/open_inwoner/scss/components/Contacts/Contacts.scss
Comment (alextreme) [reply src/open_inwoner/scss/components/Contacts/Contacts.scss:8]: Discussed, as long as this is consistent with the design this is fine

**Comment by alextreme** [comment]
Removing @annashamray , for @Bartvaderkin to double-check after Vasileios's changes

**Review by Bartvaderkin — APPROVED** [review]

**Comment by Bartvaderkin** [comment]
:heavy_check_mark: :+1:

**Pull request merged**
